jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/ViewResourceDialog.kt

```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.explorer
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.layout.panel
+import javax.swing.JComponent
+
+class ViewResourceDialog(project: Project, val resourceType: String) : DialogWrapper(project) {
+    var resourceName = ""
+    private val component by lazy {
+        panel {
+            row("$resourceType:") {
+                textField(::resourceName).constraints(grow)
```

Comment:
should we have validation if string was empty?

Reply:
Added validation for when string is empty and when string has just the prefix(in this case S3://)